### ISSUE TITLE
Placenames come on when changing basemap

### ISSUE THREAD

**Issue by mem48** [nptscot/nptscot.github.io#234]
When you switch the basemap, the placenames are enabled even if the toggle is off.

Steps to reproduce

1) Turn off place names

![image](https://github.com/user-attachments/assets/02cd7103-621d-4e63-8771-e0f4cbf9e03f)

2) Switch base map

![image](https://github.com/user-attachments/assets/4875d3f4-826d-4e47-add8-770d40b5ecd3)

Expected behavior: check the toggle and turn off placenames

Also affects https://github.com/PlaceBasedCarbonCalculator/PlaceBasedCarbonCalculator.github.io/issues/4

**Comment by Robinlovelace** [comment]
:+1: 

I would say lower priority than #228 but still important to fix this, seems like a bug.

**Comment by mem48** [comment]
I've added a fix in Carbon & Place by adding this code to the `placenames` function

```
// Listen for changes in the whole style (e.g. basemap changes)
map.on('styledata', () => {
        console.log('Toggling place names');
        const checkbox = document.getElementById('placenamescheckbox');
  	Object.entries(placenameLayers).forEach(([layerId, layer]) => {
  		map.setLayoutProperty(layerId, 'visibility', (checkbox.checked ? 'visible' : 'none'));
  	});
});
```

This works but triggers when any change to the map style is made (e.g. turning data layers on and off) rather than only when the basemap is changed. Perhaps @mvl22 can comment on if this is best practice.

**Comment by mvl22** [comment]
Thanks, Malcolm.

I've found various inefficiencies in the placenames function, in particular the way that it has to reload the `partial-style_oszoom_names.json` file each time, and the handler is re-registered, so compounds its functionality. I suspect sorting these out will eke out the underlying bug here. I'm just reworking this now.

**Comment by mvl22** [comment]
I found the problem.

The initial state was being set at `<layer>.visibility` but this should have been `<layer>.layout.visibility`. Fixed in 37a2b5f1fda9f283ee7906d76ce6007be4dba766.

In the process of tracking this down, I've also fixed dangling references in the event handler - it was actually triggering multiple times, and also the placenames .json file was being reloaded each time inefficiently. All much cleaner and more correct now.
